### PR TITLE
Minor move/rename

### PR DESCRIPTION
**Pull request by rnbwdsh**
Rename example extractors to "default" and "canned"
Move everything from __init__.py to extraction.py

Reason:
Imo __init__.py isn't a place where I'd look for logic/documentation, and the "aocd_examples_" prefix is redundant for the context of already being a param to aoce.
Also, I think "plugin" isn't the best name for one out of two plugins. I liked "hardcoded[_locations]" as well as "canned", maybe that would be an even better name? Generally extractor-name == extractor-function-name seems to be a more logical default-convention for me, because it took me a bit to understand the layer of misdirection via __init__ -> plugin ->  extract_examples(page).

### CODE REVIEW

**Comment by wimglenn** [comment]
the rename would create incompatibility with v2.0.0 https://github.com/search?q=repo%3Awimglenn%2Fadvent-of-code-data+aocd_examples_canned&type=code

**Comment by rnbwdsh** [comment]
You're the owner, and it's just out for 1 day, I think it's not too late to change it. So far, not even the google index has updated to display the 2.0.0 when you search "advent of code data pypi". Or the [Arch User Index](https://aur.archlinux.org/packages/python-aocd).

And the one crazy person who daily-updates their python packages will update it again tomorrow anyways :D
Generally I wouldn't care too much about breaking the package in any month except December.

File: aocd_example_parser/extraction.py [hunk 1]
```diff
@@ -6,6 +6,46 @@
 from aocd.examples import Page
 
 
+from aocd.examples import Example
+from aocd.examples import Page
+
+from .extraction import extract_examples
```
Comment (wimglenn) [inline aocd_example_parser/extraction.py:12]: you don't need to import it when it's in the same module

Comment (rnbwdsh) [reply aocd_example_parser/extraction.py]: fixed.

**Comment by wimglenn** [comment]
In hindsight I'll agree that the plugin names aren't great - they're already part of the entrypoint group named "adventofcode.examples" so an "aocd_examples_" prefix is even more redundant. 

What do you think about these renames? 

aocd_examples_canned -> "reference"
aocd_examples_default -> "simple"

**Comment by rnbwdsh** [comment]
Reference: def. best name
Simple: as good as "default"

**Pull request merged**